### DISCUSSION TITLE
Implementing Radar forecast data

### DISCUSSION THREAD

**Discussion by guidocioni** [open-meteo/open-meteo#816] [Ideas]
Hey @patrick-zippenfenig,
today I was thinking that it would be really cool to implement radar data (together with the forecast) in open-meteo.

This data is freely available for Germany in the open data server (radolan WN product) and allows one to make a pretty good now cast prediction of precipitation in the next 2 hours. Being based on radar this is much better than any weather model, especially in the short range.

I can foresee so many amazing projects if we have this data available: from merging together high resolution nwp models with radar to make the best now casting prediction, to train a model to learn precipitation biases for a certain location.

The data is quite easy to process (I did it myself so I have some experience): on my raspberry it takes just 10-15 seconds to process a single 2 hrs forecast. it also doesn't take a lot of space, so there's no storage issue. Plus, you really won't have to store many historical data as the archive can be wiped after a while. 

The only problem is that obviously this only has a value if it is updated often (every 5 mins) and I don't know if the infrastructure you designed is supposed to have jobs ran so often.

This could be a really good preparation to the day when OPERA will finally open the archive of the European radar composite, as we could do this not just for Germany but for all European countries! 

What do you think?

**Comment by patrick-zippenfenig** [comment]
I already thought about that. Good coverage is an issue. Data is available for some European countries and North America.

As Open-Meteo is a member of the user group of the RODEO project which aims to provide a [radar composite as open-data](https://rodeo-project.eu/projectactivities/#weather-radar-data), I hope to get access sooner than later :)

**Reply by guidocioni** [reply]
Ah ok, so that's the same composite I was mentioning (OPERA). So the current strategy is to wait for access to European composite instead than trying to build something only for Germany at the beginning?
Good to know that.
I guess the final goal would be similar to what I suggested in the first message?

**Reply by patrick-zippenfenig** [reply]
I am unsure about the OPERA timeline. Starting with DWD might be an option.

Otherwise, yes, it can be converted into a time-series database including archives and accessed for individual points (or ideally small areas)

**Reply by guidocioni** [reply]
so, just for me to understand, you're planning on implementing a German radar reader or that's not even on the Plan? 😄

**Reply by patrick-zippenfenig** [reply]
Not sure. Depends on available time, interest from other users and if there are any news regarding OPERA open-data availability.

**Reply by guidocioni** [reply]
Regarding Opera I don't see anything moving in the next year or so. I'd like to contribute to this feature but unfortunately Swift is not in the realm of my languages 😕

**Comment by patrick-zippenfenig** [comment]
Quick update: DWD will switch radar products to [HDF5 in October](https://www.dwd.de/DE/leistungen/opendata/neuigkeiten/opendata_juli2024_2.html). The Rodeo working group will also meet end of September, therefore I will wait a bit longer before starting to integrate the old RADOLAN format.
